src/jit/gentree.cpp

```diff
@@ -12637,7 +12637,7 @@ GenTree* Compiler::gtFoldExprSpecial(GenTree* tree)
                 {
                     // The tree under the box must be side effect free
                     // since we will drop it if we optimize.
-                    assert(!gtTreeHasSideEffects(op->gtBox.AsOp()->gtOp1, GTF_SIDE_EFFECT));
+                    assert(!gtTreeHasSideEffects(op->AsBox()->AsOp()->gtOp1, GTF_SIDE_EFFECT));
```

Comment:
There is an extra (pre-existed) cast here, could you please fix it:
`op->AsBox()->AsOp()->gtOp1` should be  `op->AsBox()->BoxOp()`.